src/lib/client.js

```diff
@@ -70,9 +76,12 @@ class ModularClient extends Client {
 			partials: [Partials.Channel],
 		});
 		this.#config = config;
+
+		this.#waitingJob = new cron.CronJob('0 0 * * *', () => this.#createWaitingReport(config));
```

Comment:
Bind the current object to the method (this allows us to access private properties in the callback)

```suggestion
		this.#waitingJob = new cron.CronJob('0 0 * * *', this.#createWaitingReport.bind(this));
```